### PR TITLE
Enable configuration via envvars

### PR DESCRIPTION
**Pull request by ChrisPRobinson**

### CODE REVIEW

**Comment by ChrisPRobinson** [comment]
Need to add tests. Enabling to make some configuration setting easier

File: lib/etl/config.rb [hunk 1]
```diff
@@ -3,72 +3,165 @@
...
       @db
     end
-    
+
+    def database_env_vars
+      conn_params = {}
+      conn_params[:encoding] = "utf8"
+      conn_params[:reconnect] = false
+      conn_params[:pool] = 5
+      conn_params[:adapter] = ENV.fetch('ETL_DATABASE_DB_NAME', 'postgres')
```
Comment (ChrisPRobinson) [inline lib/etl/config.rb]: Env var is wrong for this one

File: lib/etl/config.rb [hunk 2]
```diff
@@ -3,72 +3,169 @@
...
     def db_file
       @config_dir + "/database.yml"
     end
-    
+
     def db(&b)
-      @db ||= self.class.load_file(db_file)
+      get_envvars = is_true_value(ENV.fetch('ETL_DATABASE_ENVVARS', false))
+      @db ||= if get_envvars
+                @db = database_env_vars
```
Comment (emya) [inline lib/etl/config.rb]: just `database_env_vars` ?

File: lib/etl/config.rb [hunk 3]
```diff
@@ -3,72 +3,169 @@
...
+              else
+                @db ||= self.class.load_file(db_file)
```
Comment (emya) [inline lib/etl/config.rb]: just `self.class.load_file(db_file)` ?

File: lib/etl/config.rb [hunk 4]
```diff
@@ -3,72 +3,169 @@
...
+
     def aws(&b)
-      @aws ||= self.class.load_file(aws_file)
+      get_envvars = ENV.fetch('ETL_AWS_ENVVARS', false)
+      @aws ||= if is_true_value(get_envvars)
+                @aws = {}
+                @aws[:aws_region] = ENV.fetch('ETL_AWS_REGION', 'us-west-2')
+                @aws[:s3_bucket] = ENV.fetch('ETL_AWS_S3_BUCKET')
+                @aws[:role_arn] = ENV.fetch('ETL_AWS_ROLE_ARN')
+              else
```
Comment (emya) [inline lib/etl/config.rb]: I think `@aws` gonna be the value of `ENV.fetch('ETL_AWS_ROLE_ARN')`.
```
h = {}
h[] = ENV.fetch()
h
```
To assign the hash to `@aws`

File: lib/etl/config.rb [hunk 5]
```diff
@@ -3,72 +3,169 @@
...
+                      @redshift = {}
+                      @redshift[:user] = ENV.fetch('ETL_REDSHIFT_USER', 'masteruser')
+                      @redshift[:password] = ENV.fetch('ETL_REDSHIFT_PASSWORD', 'root')
+                      if !use_odbc_dsn_connection
+                        @redshift[:dbname] = ENV.fetch('ETL_REDSHIFT_DB_NAME', 'dev')
+                        @redshift[:host] = ENV.fetch('ETL_RESHIFT_HOST')
+                        @redshift[:port] = ENV.fetch('ETL_REDSHIFT_PORT', 5439)
+                      else
+                        @redshift[:port] = ENV.fetch('ETL_REDSHIFT_DSN', 'MyRealRedshift')
+                      end
```
Comment (emya) [inline lib/etl/config.rb]: Need to put the hash to assign to `@redshift`

File: lib/etl/config.rb [hunk 6]
```diff
@@ -3,72 +3,169 @@
...
+
+                c[:metrics] = {}
+                c[:metrics][:class] = ENV.fetch('ETL_METRICS_CLASS', "ETL::Metrics")
+                c[:metrics][:file] = ENV.fetch('ETL_METRICS_FILE_PATH', '/tmp/etl-metrics.log')
+                c[:metrics][:series] = 'etlv2_job_run'
+
+                c[:slack] = {}
+                c[:slack][:url] = ENV.fetch('ETL_SLACK_URL')
+                c[:slack][:channel] = ENV.fetch('ETL_SLACK_CHANNEL')
+                c[:slack][:username] = ENV.fetch('ETL_SLACK_USERNAME')
```
Comment (emya) [inline lib/etl/config.rb]: return `c`

**Comment by ChrisPRobinson** [comment]
@emya thanks for the review, totally missed the logic around the ifs, still getting used to writing code that initializes off of the ifs.

**Pull request merged**